funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -658,7 +676,8 @@ def submit(self, func, *args, container_id: str = 'RAW', task_id: str = None, **
                        ser_code + ser_params)
 
         self.submit_raw(payload.pack())
-        self.tasks[task_id] = Future()
+        self.tasks[task_id] = HTEXFuture(self)
+        self.tasks[task_id].task_id = task_id
```

Comment:
There's a very slight race condition here I think (but I've seen this sort of thing trigger in parsl in the DFK when tasks complete fast). I think the task can complete any time including-and-after line 678, before self.tasks[task_id] is set.

Reply:
(this is also true before this PR #606)

Reply:
crossref https://github.com/Parsl/parsl/issues/2033

Reply:
ok this is confusing me. This codepath isn't reached when using the funcx endpoint as far as I can see. Is `def submit()` just here for if you're using the htex component on your local machine without the rest of funcx?